File: spring-boot-starter-breuninger-jobs/src/main/java/com/breuninger/boot/jobs/eventbus/EventBusConfiguration.java
package com.breuninger.boot.jobs.eventbus;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import com.breuninger.boot.jobs.service.JobService;

@Configuration
public class EventBusConfiguration {

  private final JobService jobService;

  public EventBusConfiguration(final JobService jobService) {
    this.jobService = jobService;
  }

  @Bean
  public JobStateChangeListener logJobEventListener() {
    return new LogJobStateChangeListener();
  }

  @Bean
  public JobStateChangeListener persistenceJobEventListener() {
    return new PersistenceJobStateChangeListener(jobService);
  }
}

File: jobs/src/main/kotlin/com/breuninger/boot/jobs/eventbus/autoconfigure/EventBusConfiguration.kt
package com.breuninger.boot.jobs.eventbus.autoconfigure

import com.breuninger.boot.jobs.eventbus.service.LogJobExecutionStateChangedEventListener
import com.breuninger.boot.jobs.eventbus.service.MeterJobExecutionStateChangedEventListener
import com.breuninger.boot.jobs.eventbus.service.PersistJobExecutionStateChangedEventListener
import com.breuninger.boot.jobs.service.JobExecutionService
import io.micrometer.core.instrument.MeterRegistry
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
@ConditionalOnProperty(prefix = "breuni.jobs", name = ["enabled"], havingValue = "true")
class EventBusConfiguration {

  @Bean
  fun logJobExecutionStateChangedEventListener() = LogJobExecutionStateChangedEventListener()

  @Bean
  fun persistJobExecutionStateChangedEventListener(jobExecutionService: JobExecutionService) =
    PersistJobExecutionStateChangedEventListener(jobExecutionService)

  @Bean
  fun meterJobExecutionStateChangedEventListener(meterRegistry: MeterRegistry) =
    MeterJobExecutionStateChangedEventListener(meterRegistry)
}
